androidshared/src/main/java/org/odk/collect/androidshared/sqlite/DatabaseMigrator.java

```diff
@@ -7,5 +7,11 @@ public interface DatabaseMigrator {
 
     void onUpgrade(SQLiteDatabase db, int oldVersion);
 
+    /**
+     * Downgrading an app is generally not officially supported without first uninstalling
+     * the current version. However, it may be possible in some cases, depending on the app
+     * and device. Rooted devices, for example, might allow more direct downgrades.
```

Comment:
Thanks for looking into this! Seeing this, I'm actually leaning towards the oppositive and just exploding if we get a call to `onDowngrade`. I think the idea that data gets cleared out might be more dangerous than just crashing and letting the user export the data if they still need it or at least understand that they're in a bad state. Would be good to get input from @lognaturel on this as well.

Reply:
Interesting let's see what @lognaturel thinks.